queue/adaptive_order_storage.go

```diff
@@ -57,6 +64,13 @@ func (items *adaptiveOrderItems) updateWaiting(ctx context.Context) {
 			continue
 		}
 		status := job.Status()
+		ti := job.TimeInfo()
+
+		if ti.WaitUntil.After(now) {
+			new = append(new, id)
```

Comment:
Is the inverse possible, that is, some item with an expired WaitUntil had its WaitUntil updated to be later? Or are items' WaitUntils fixed once they enter the queue? If the former, then you should also check items not in items.waiting? If the latter, then this is fine.

Reply:
in general, jobs aren't mutable after the enter the queue, though for the non-remote queues it's just a pointer and it can be modified through the normal means. 

The adaptive order queue, reexamines the "blocked/waiting" queue, intermittently in the background to avoid stranding jobs that could be worked on in the queue. I think its fine in this case.